spring-ai-docs/src/main/antora/modules/ROOT/pages/api/mcp/mcp-annotations-server.adoc

```diff
@@ -302,53 +302,296 @@ public CompleteResult completeCode(String prefix) {
+public String unifiedTool(
+        McpSyncRequestContext context,
         @McpToolParam(description = "Input", required = true) String input) {
     
-    // Access server exchange features
-    exchange.loggingNotification(...);
-    exchange.progressNotification(...);
-    exchange.ping();
+    // Access request and metadata
+    String progressToken = context.request().progressToken();
+    
+    // Logging with convenient methods
+    context.info("Processing: " + input);
+    
+    // Progress notifications
+    if (progressToken != null) {
+        context.progress(50); // Simple percentage
+    }
+    
+    // Ping client
+    context.ping();
     
-    // Can call client methods
-    CreateMessageResult result = exchange.createMessage(...);
-    ElicitResult elicitResult = exchange.createElicitation(...);
+    // Check capabilities before using
+    if (context.elicitEnabled()) {
+        // Request user input (only in stateful mode)
+        StructuredElicitResult<UserInfo> elicitResult = context.elicit(UserInfo.class);
+        if (elicitResult.action() == ElicitResult.Action.ACCEPT) {
+            // Use elicited data
+        }
+    }
+    
+    if (context.sampleEnabled()) {
+        // Request LLM sampling (only in stateful mode)
+        CreateMessageResult samplingResult = context.sample("Generate response");
+        // Use sampling result
+    }
     
-    return "Processed with full context";
+    return "Processed with unified context";
 }
 ----
 
-=== Stateless (with McpTransportContext or without)
+=== Simple Operations (No Context)
 
-Stateless implementations are simpler and don't require server exchange:
+For simple operations, you can omit context parameters entirely:
 
 [source,java]
 ----
-@McpTool(name = "stateless-tool", description = "Simple stateless tool")
+@McpTool(name = "simple-add", description = "Simple addition")
 public int simpleAdd(
         @McpToolParam(description = "First number", required = true) int a,
         @McpToolParam(description = "Second number", required = true) int b) {
     return a + b;
 }
+----
+
+=== Lightweight Stateless (with McpTransportContext)
+
+For stateless operations where you need minimal transport context:
 
-// With transport context if needed
-@McpTool(name = "stateless-with-context", description = "Stateless with context")
-public String withContext(
+[source,java]
+----
+@McpTool(name = "stateless-tool", description = "Stateless with transport context")
+public String statelessTool(
         McpTransportContext context,
         @McpToolParam(description = "Input", required = true) String input) {
-    // Limited context access
+    // Access transport-level context only
+    // No bidirectional operations (roots, elicitation, sampling)
     return "Processed: " + input;
 }
 ----
 
+[IMPORTANT]
+**Stateless servers do not support bidirectional operations:**
+
+* `roots()` - Not available
+* `elicit()` - Not available  
```

Comment:
I think those should be named `elicitation()` and `sampling()` (same for the Enabled variations below)